modules/swagger-codegen/src/main/java/io/swagger/codegen/languages/NodeJSServerCodegen.java

```diff
@@ -164,7 +164,7 @@ public String apiFilename(String templateName, String tag) {
     }
 
     private String implFileFolder(String output) {
-        return outputFolder + "/" + output + "/" + apiPackage().replace('.', '/');
+        return (outputFolder + File.separator + output + File.separator + apiPackage()).replace('.', File.separatorChar);
```

Comment:
1. Is there a file path API for Java to avoid such an error-prone path concatenation implementation?
2. Why should all the dots (`.`) be replaced with a path separator? What if I have a folder with a dot in its name?

Reply:
1. Not that I'm aware of and I agree the current solution is error-prone so I'm open to new ideas for a better implementation.

2. I do not know the reason. The logic was added before I joined this project. My suggestion is to keep the current behavior and try to come up with workaround/solution if users file an issue about it.

Reply:
1. OK
2. If you want to keep the current behavior, you should probably drop those parentheses to limit the replacement only for the `apiPackage()` string instead of the whole result:

    ```
    outputFolder + File.separator + output + File.separator + apiPackage().replace('.', File.separatorChar)
    ```

Reply:
2. Done via e10c09d